scripts/generate_config_checks.py

```diff
@@ -3,19 +3,42 @@
 """Generate C preprocessor code to check for bad configurations.
 """
 
+from typing import Iterator
+
 import framework_scripts_path # pylint: disable=unused-import
 from mbedtls_framework.config_checks_generator import * \
     #pylint: disable=wildcard-import,unused-wildcard-import
+from mbedtls_framework import config_history
+
+ALWAYS_ENABLED_SINCE_1_0 = frozenset([
+    'MBEDTLS_PSA_CRYPTO_CONFIG',
+    'MBEDTLS_USE_PSA_CRYPTO',
+])
+
+def checkers_for_removed_options() -> Iterator[Checker]:
+    """Discover removed options. Yield corresponding checkers."""
+    history = config_history.ConfigHistory()
+    old_public = history.options('mbedtls', '3.6')
+    new_public = (history.options('tfpsacrypto', '1.0') |
+                  history.options('mbedtls', '4.0'))
+    internal = history.internal('tfpsacrypto', '1.0')
+    for option in sorted(old_public - new_public):
```

Comment:
More pythonic but maybe harder to read? 

~~~~~~
yield from (
    Internal(option) if option in internal else Removed(option, 'TF-PSA-Crypto 1.0')
    for option in sorted(sorted(old_public - new_public))
)
~~~~~~

Reply:
Yeah I don't think we should aim for maximal code density. Actually I was even on the verge of suggesting